### PR TITLE
Bump min pyglet to 2.0.6 for RGBA caret support

### PR DESCRIPTION
**Pull request by pushfoo**
### Changes

* Bump the minimum pyglet version to 2.0.6 so we can guarantee RGBA caret support in labels

### How to test

1. `pytest tests/unit` as usual
2. Run sound examples locally?

I've held off on further action on #1690 until pyglet shipped https://github.com/pyglet/pyglet/pull/806.

### Potential Issue

The one potential issue with this PR is an ignored exception related to OpenAL:
```console
============================= 558 passed, 2 xfailed, 2 warnings in 13.40s ==============================
Exception ignored in: <function OpenALDriver.__del__ at 0x7fabdc82bc10>
Traceback (most recent call last):
  File "/home/user/src/arcade/.venv/lib/python3.9/site-packages/pyglet/media/drivers/openal/adaptation.py", line 27, in __del__
    self.delete()
  File "/home/user/src/arcade/.venv/lib/python3.9/site-packages/pyglet/media/drivers/openal/adaptation.py", line 34, in delete
    self.worker.stop()
  File "/home/user/src/arcade/.venv/lib/python3.9/site-packages/pyglet/media/mediathreads.py", line 59, in stop
    self._thread.join()
  File "/usr/lib/python3.9/threading.py", line 1030, in join
    raise RuntimeError("cannot join current thread")
RuntimeError: cannot join current thread
```

System info:
```console
$ python -m arcade

Arcade 3.0.0.dev21
------------------
vendor: Intel
renderer: Mesa Intel(R) UHD Graphics 620 (KBL GT2)
version: (4, 6)
python: 3.9.2 (default, Feb 28 2021, 17:03:44) 
[GCC 10.2.1 20210110]
platform: linux
pyglet version: 2.0.6
PIL version: 9.4.0
```

### CODE REVIEW

**Comment by pushfoo** [comment]
There may be an issue with text under pyglet 2.0.6 on Linux. I'm converting this into a draft until the issue is investigated further.

**Comment by pushfoo** [comment]
tl;dr the issue (https://github.com/pyglet/pyglet/issues/823) doesn't seem to affect arcade's text, but we may want to hold off on further use of pyglet shapes

1. We seem to be unaffected by the issue because our text widgets do not use `pyglet.gui.widgets.TextEntry`, nor do they use Pyglet's shapes much directly
2. The issue has been present since [shortly after pyglet 2.0.2 (pyglet commit hash `44da1724539823ac`)](https://github.com/pyglet/pyglet/tree/44da1724539823acb8f40fe0b62b378b31e6a0e8)
3. A user on the pyglet Discord is investigating it further
